app/code/Magento/Catalog/Controller/Adminhtml/Product/Save.php

```diff
@@ -117,6 +117,7 @@ public function execute()
                 $productId = $product->getEntityId();
                 $productAttributeSetId = $product->getAttributeSetId();
                 $productTypeId = $product->getTypeId();
+                $this->getRequest()->setParam('type', $productTypeId);
```

Comment:
Did you check the problem? problem is magento not getting the type from product if it is set in url.